tests/conftest.py

```diff
@@ -323,6 +326,31 @@ def ptfhost(ansible_adhoc, tbinfo, duthost):
         ptf_host = duthost.host.options["inventory_manager"].get_host(duthost.hostname).get_vars()["ptf_host"]
         return PTFHost(ansible_adhoc, ptf_host)
 
+@pytest.fixture(scope="module")
+def k8shosts(ansible_adhoc, request, creds):
+    """
+    Shortcut fixture for getting Kubernetes hosts
+    """
+    master_id = request.config.getoption("--kube_master") 
+    master_id_elements = master_id.split('_')
+    assert len(master_id_elements) == 2, "Invalid master identifier format. Please use format {servernumber}_{mastersetnumber}"
+    master_server_id = master_id_elements[0]
+    master_set_id = master_id_elements[1]
+    k8s_ansible_group = "k8s_vms{}_{}".format(master_set_id, master_server_id)
+    master_vms = {}
+    inv_files = request.config.getoption("ansible_inventory")
+    for inv_file in inv_files:
+        if "k8s" in inv_file:
+            k8s_inv_file = inv_file
+    with open('../ansible/{}'.format(k8s_inv_file), 'r') as kinv:
+        k8sinventory = yaml.safe_load(kinv)
```

Comment:
Can you consider using Ansible's inventory manager to parse the inventory file? Examples:
* https://github.com/Azure/sonic-mgmt/blob/master/ansible/library/testbed_vm_info.py#L91
* https://github.com/Azure/sonic-mgmt/blob/master/tests/conftest.py#L502

Reply:
I will look into this, thanks for the examples

Reply:
From what I can tell, Ansible's InventoryManager is good for getting a specific host's data when given the hostname
https://github.com/ansible/ansible/blob/devel/lib/ansible/inventory/manager.py#L194
In the InventoryManager class, I don't see a direct way to get all hosts given group name like k8s_vms1_21

For preexisting InventoryManager usage in sonic-mgmt repo, it looks like the specific hostname SONiC hostname is directly retrieved from the testbed info, and specific EOS hostname is constructed after loading the topology yaml.

Reply:
I am not quit sure if you can get something from the `groups` property: https://github.com/ansible/ansible/blob/devel/lib/ansible/inventory/manager.py#L174
If it is too complicated to use InventoryManager, I am also OK with loading yaml. The drawback is that this again assumes that the `k8s_master_ansible_group` must be a top level key in the inventory yaml.

Reply:
I noticed the `groups` property`. I looked for some way to specify group name and retrieve associated hosts, but I could not find it

Reply:
OK, directly loading yaml is fine too.